tests/sentry/testutils/test_collection_modify_items.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable, cast
+
+import pytest
+
+from sentry.testutils.pytest.sentry import _get_keep_and_discard_items
+
+
+@dataclass(frozen=True)
+class MockItem:
+    nodeid: str
+
+
+def _assert_shuffled(
+    keep: list[pytest.Item], discard: list[pytest.Item], items: list[pytest.Item]
+) -> None:
+    assert len(keep) == len(items)
+    assert set(keep) == set(items)
+    out_of_order = False
+    for i in range(len(items)):
+        if keep[i] != items[i]:
+            out_of_order = True
+    assert out_of_order
+
+
+@pytest.mark.parametrize(
+    "num_items,total_groups,current_group,grouping_strategy,shuffle_tests,shuffler_seed,sample_rate,num_expected_keep_items,check_function",
```

Comment:
this is a sign that the test does too much -- this is impossible to tell what you're parametrizing from the list of testcases

also prefer the not-stringly-typed tuple of strings instead of giant string here